drift/lib/src/runtime/manager/references.dart

```diff
@@ -288,25 +285,70 @@ T _defaultStateTransformer<
   return $state;
 }
 
-/// Type definition for a function which transforms a List of TypedResults
-typedef TypedResultTransformer = Future<List<TypedResult>> Function(
-    List<TypedResult> results);
-
 /// When a user requests that certain fields are prefetched, we create a [PrefetchHooks] class for the manager.
 /// This class has hooks for adding joins to the query before the query is executed, and for running prefetches after the query is executed.
 /// {@macro manager_internal_use_only}
 class PrefetchHooks {
   /// This callback is used to add joins to the query before it is executed.
   late final StateTransformer withJoins;
 
-  /// This callback is used to prefetch referenced data and insert it into the TypedResult object.
-  late final TypedResultTransformer withPrefetches;
+  /// A function which will return list of streams which for each prefetch data source.
+  final List<Stream<List<MultiTypedResultEntry>>> Function(List<TypedResult>,
+      {required bool watch})? prefetchedDataStreamsCallback;
 
   /// Create a [PrefetchHooks] object
   PrefetchHooks(
-      {StateTransformer? addJoins, TypedResultTransformer? withPrefetches}) {
+      {StateTransformer? addJoins, this.prefetchedDataStreamsCallback}) {
     withJoins = addJoins ?? _defaultStateTransformer;
-    this.withPrefetches = withPrefetches ?? (results) async => results;
+  }
+
+  /// Internal function for injecting the prefetched data into the TypedResult object.
+  ///
+  /// When [watch] is true, the streams will be watched and the TypedResult object will be updated when the prefetched data changes.
+  /// Otherwise, the streams of prefetched data will only be read once.
+  Stream<List<TypedResult>> _addPrefetchedDataToStream(
+      Stream<List<TypedResult>> itemStream,
+      {required bool watch}) {
+    /// If this table contains no reverse references, we can just return the stream as is.
+    if (prefetchedDataStreamsCallback == null) {
+      return itemStream;
+    }
+
+    /// Otherwise we add the prefetched data to the stream
+    return itemStream.asyncMap((rows) {
```

Comment:
This is probably super tricky, but at the moment the selects for the streams are not running in a transaction right?

I wonder if we should just register a stream depending on all the tables (including possible prefetches) and then run that in a transaction at once instead of combining independent query streams here?
Not being consistent with `get` in that we don't use a transaction here is not great, but we also don't have the fundamentals in place to get this right at the moment. I'll take a look at adding those first, it's just something remember.

Reply:
I don't have the slightest idea on how to get this working with a transaction.

How would having the other operations as a `get` help?